doc/man7/password-encoding.pod

```diff
@@ -0,0 +1,134 @@
+opportunistically prepends each byte with a zero byte to obtain the UCS-2
+encoding of the characters, which it stores as a BMPString.
+
+Note that since there is no check of your locale, this may produce UCS-2 /
+UTF-16 characters that do not correspond to the original pass phrase characters
+for other character sets, such as any ISO-8859-X other than ISO-8859-1.
+
+=back
+
+OpenSSL versions older than 1.1.0 do variant 2 only, and that is the reason why
+OpenSSL still does this, to be able to read files produced with older versions.
+
+It should be noted that this approach isn't entirely fault free.
+A passphrase encoded in ISO-8859-1 could very well have a sequence such as
+0xC3 0xAF (which is the two characters "LATIN CAPITAL LETTER A WITH TILDE"
+and "MACRON" in ISO-8859-1 encoding), but would be misunterpreted as the
+perfectly valid UTF-8 encoded U+00EF (LATIN SMALL LETTER I WITH DIARESIS).
+
+=head2 OSSL_STORE
+
+L<ossl_store(7)> acts as a general interface to access all kinds of objects,
+potentially protected with a pass phrase, a PIN or something else.
+This API currently doesn't stipulate any specific encoding of pass phrases, but
+uses the underlying routines with their behaviours.
+This means that when using the built-in C<file:> scheme loader, the pass phrase
+to unlock a PKCS#12 file will be treated as described for PKCS#12 above, and
+the pass phrase for a PEM files will be treated as the general case described
+above, since that loader uses the same underlying routines.
+I<Note that other loaders will have their own behaviours>.
+
+=head1 RECOMMENDATIONS
+
+For creating new pass phrase protected objects, make sure the pass phrase is
+encoded using UTF-8.  This is default on most modern Unixes, but may involve an
+effort on other platforms.
+
+For opening pass phrase protected objects where you know what character
+encoding was used for the encryption pass phrase, make sure to use the same
+encoding again. This will ensure that the exact same byte sequence is used in
+both cases.  In case the character encoding is UTF-8, the exact locale is not
+important, as long as it's still UTF-8.
+
+For opening pass phrase protected objects where the character encoding that was
+used is unknown, or where the producing application is unknown, try one of the
+following:
+
+=over 4
+
+=item 1.
+
+Convert the pass phrase to UTF-8 and try with the result.  Specifically with
+PKCS#12, this should open up any object that was created according to spec.
+
+=item 2.
+
+Do a naïve (i.e. purely mathematical) ISO-8859-1 to UTF-8 conversion and try
+with the result.  This differs from the previous attempt because ISO-8859-1
+maps directly to U+0000 to U+00FF, which other non-UTF-8 character sets do
+not.
+
+=item 3.
+
+Try the password (i.e. the byte sequence) that you have as it is, without
+conversion.  It's possible that it's the exact right sequence.  This probably
+only works with objects created with OpenSSL older than 1.1.0.
```

Comment:
https://github.com/openssl/openssl/pull/6179#discussion_r186398499. As far as "older than 1.1.0" goes, it's previous item.

Reply:
One can wonder if this 3rd item is needed at all. Indeed, how does user end up needing this recommendations? Pass phrase didn't work, right? But doesn't this mean that "as it is" has already been tried? By assuming that encoding was known and same as original used to protect the object. But even if it's required, wouldn't it be more natural to place it first? I mean isn't assumption that encoding didn't change most natural one?